src/coreclr/jit/codegenlinear.cpp

```diff
@@ -2599,32 +2599,17 @@ void CodeGen::genStoreLongLclVar(GenTree* treeNode)
     assert(!varDsc->lvPromoted);
     GenTree* op1 = treeNode->AsOp()->gtOp1;
 
-    // A GT_LONG is always contained, so it cannot have RELOAD or COPY inserted between it and its consumer,
-    // but a MUL_LONG may.
-    noway_assert(op1->OperIs(GT_LONG) || op1->gtSkipReloadOrCopy()->OperIs(GT_MUL_LONG));
+    // A GT_LONG is always contained so it cannot have RELOAD or COPY inserted between it and its consumer.
```

Comment:
Need to update the comment about "RELOAD or COPY"?

Reply:
I believe it is accurate? We are asserting below what it says, that we have a contained `GT_LONG` (the "contained" part is asserted indirectly as the comment states).

Reply:
The way I understand is we do not assert this part of the comment "it cannot have RELOAD or COPY inserted between it and its consumer"?

Reply:
We do: `op1` is the direct operand, `treeNode` is the consumer. Were a reload/copy make its way into this code, it would become `op1`, making the assert fail.

Reply:
ah. Yes.